modules/basics/src/main/java/com/opengamma/strata/basics/location/Country.java

```diff
@@ -41,10 +42,12 @@ public final class Country
   /**
    * Loads the ISO alpha-2 and alpha-3 country codes into a bidirectional map.
    */
-  private static final Supplier<ImmutableBiMap<String, String>> COUNTRY_CODES = Suppliers.memoize(() ->
-      ImmutableBiMap.copyOf(PropertiesFile.of(
-          ResourceLocator.ofClasspath(Country.class, "country.properties").getCharSource())
-          .getProperties().asMap()));
+  private static final Supplier<ImmutableBiMap<String, Country>> COUNTRY_CODES = Suppliers.memoize(() ->
+      ImmutableBiMap.copyOf(MapStream.of(PropertiesFile.of(
+                  ResourceLocator.ofClasspath(Country.class, "country.properties").getCharSource())
+              .getProperties().asMap())
+          .mapValues(Country::of)
```

Comment:
I believe this will cause all instances in the config file to be added to CACHE, thus I don't think you need to use `concatToSet` with the CACHE in `getAvailableCountries`

Reply:
This is correct, but EU is not in the config file, and by continuing to concat with the CACHE it means that any new countries that a user adds with `Countries::of` will also become available in `getAvailableCountries`. 

I can't just return the CACHE either, as that would rely on the Supplier having been called at least once already.

Reply:
Write a test, but it looks to me that all instances are constructed via `of(String)`, where `computeIfAbsent()` will always call the supplier. Thus the CACHE will always contain all instances.

Reply:
That's correct, but the ones coming from the config file don't run until the `Suppliers.memoize()` thing is called for the first time. So to ensure that we get all countries, we need to make sure that `COUNTRY_CODES.get()` has been called. 
```
  @Test
  public void test_whether_Cache_is_always_fully_populated() {
    Set<Country> available = Country.getAvailableCountries(); //.size() == 46
    COUNTRY_CODES.get();
    Set<Country> more = Country.getAvailableCountries(); //.size() == 252
    assertThat(available.size()).isEqualTo(more.size()); // fails
  }
```

Reply:
OK then, so the `concatToSet` is fine, but the result does need sorting

Reply:
`concatToSet` is not really needed now I think about it, as long as we call `COUNTRY_CODES.get()` the CACHE will contain the complete list. I'm then taking the values from the CACHE, and creating a SortedSet as the result.